tensorflow/lite/micro/kernels/ceva_bx1/fully_connected.cc

```diff
@@ -0,0 +1,348 @@
+/* Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "tensorflow/lite/kernels/internal/reference/fully_connected.h"
+
+#include "CEVA_TFLM_lib.h"
+#include "tensorflow/lite/c/builtin_op_data.h"
+#include "tensorflow/lite/c/common.h"
+#include "tensorflow/lite/kernels/internal/common.h"
+#include "tensorflow/lite/kernels/internal/quantization_util.h"
+#include "tensorflow/lite/kernels/internal/reference/integer_ops/fully_connected.h"
+#include "tensorflow/lite/kernels/internal/tensor_ctypes.h"
+#include "tensorflow/lite/kernels/kernel_util.h"
+#include "tensorflow/lite/micro/kernels/kernel_util.h"
+#ifdef MCPS_MEASUREMENT
+#include "..\CEVA_TFLM_lib\Src\header\mcps_macros.h"
+#endif
+
+namespace tflite {
+namespace {
+
+struct OpData {
+  // The scaling factor from input to output (aka the 'real multiplier') can
+  // be represented as a fixed point multiplier plus a left shift.
+  int32_t output_multiplier;
+  int output_shift;
+  // The range of the fused activation layer. For example for kNone and
+  // uint8_t these would be 0 and 255.
+  int32_t output_activation_min;
+  int32_t output_activation_max;
+  // The index of the temporary tensor where the quantized inputs are cached.
+  int input_quantized_index;
+  // Cached zero point values of tensors.
+  int32_t input_zero_point;
+  int32_t filter_zero_point;
+  int32_t output_zero_point;
+};

```

Comment:
let's use the shared struct:
https://github.com/tensorflow/tensorflow/blob/9d6681202e062fbf6df8bcb39d70b129333f590c/tensorflow/lite/micro/kernels/fully_connected.h#L26-L41

and also shared code (as much as possible):
https://github.com/tensorflow/tensorflow/blob/9d6681202e062fbf6df8bcb39d70b129333f590c/tensorflow/lite/micro/kernels/fully_connected_common.cc

See https://github.com/tensorflow/tensorflow/pull/46242 for an example of the direction that we would like to move towards in terms of sharing code between the reference and optimized implementations.